dotnet/InlineSecretsProvider/Program.cs

```diff
@@ -0,0 +1,148 @@
+                var @object = new Pulumi.Aws.S3.BucketObject(
+                    "index",
+                    new Pulumi.Aws.S3.BucketObjectArgs
+                    {
+                        Bucket = siteBucket.BucketName, // reference to the s3 bucket object
+                        Content = indexContent,
+                        Key = "index.html", // set the key of the object
+                        ContentType = "text/html; charset=utf-8", // set the MIME type of the file
+                    });
+
+                var bucketPolicyDocument = siteBucket.Arn.Apply(bucketArn =>
+                {
+                    return Output.Create(Pulumi.Aws.Iam.GetPolicyDocument.InvokeAsync(
+                        new Pulumi.Aws.Iam.GetPolicyDocumentArgs
+                        {
+                            Statements = new List<Pulumi.Aws.Iam.Inputs.GetPolicyDocumentStatementArgs>
+                            {
+                                new Pulumi.Aws.Iam.Inputs.GetPolicyDocumentStatementArgs
+                                {
+                                    Effect = "Allow",
+                                    Principals = new List<Pulumi.Aws.Iam.Inputs.GetPolicyDocumentStatementPrincipalArgs>
+                                    {
+                                        new Pulumi.Aws.Iam.Inputs.GetPolicyDocumentStatementPrincipalArgs
+                                        {
+                                            Identifiers = new List<string> { "*" },
+                                            Type = "AWS",
+                                        },
+                                    },
+                                    Actions = new List<string> { "s3:GetObject" },
+                                    Resources = new List<string> { $"{bucketArn}/*" },
+                                },
+                            },
+                        }));
+                });
+
+                // set the access policy for the bucket so all objects are readable
+                new Pulumi.Aws.S3.BucketPolicy(
+                    "bucket-policy",
+                    new Pulumi.Aws.S3.BucketPolicyArgs
+                    {
+                        Bucket = siteBucket.BucketName,
+                        Policy = bucketPolicyDocument.Apply(x => x.Json),
+                    });
+
+                // export the website url
+                return new Dictionary<string, object?>
+                {
+                    ["secret"] = Pulumi.Output.CreateSecret("hello world"),
+                    ["website_url"] = siteBucket.WebsiteEndpoint,
+                };
+            });
+
+            // to destroy our program, we can run "dotnet run destroy"
+            var destroy = args.Any() && args[0] == "destroy";
+
+            var projectName = "inline_s3_project";
+            var stackName = "dev";
+            var secretsProvider = "awskms://aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee?region=us-west-2";
```

Comment:
README should be updated to include changing this value in the code. Might even make it a little more bullet proof by adding a janky check on the next line:

```cs
if (secretsProvider === "awskms://aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee?region=us-west-2") {
  // throw exception asking user to change the value. 
}
```

In addition, this likely makes it difficult to test no? To support that we might need to support reading this through an env var. But that can probably be left as a follow up.

Reply:
Good suggestion. Didn't want to leave this 'dangling' so I added that and the CI setups.